generate-new/src/Render/Spec/Extends.hs

```diff
@@ -187,9 +187,18 @@ classes Spec {..} = do
       class PeekChain (xs :: [Type])
       class PokeChain (xs :: [Type])
       type family Extends (p :: [Type] -> Type) (x :: Type) :: Constraint where ..
+
+      -- | We don't really need constraint units produced by `Extends`, so this type
+      -- family will ensure that it would reduce and drop the result
+      --
+      -- That will result in less overhead because `Extendss` reduces into a single
+      -- contraint unit `()` instead of cons-list `((), ((), ()))` produced by `(,)`
+      type family SeqConstraintUnit (a :: Constraint) (b :: Constraint) :: Constraint where
+        SeqConstraintUnit () b = b
+      
       type family Extendss (p :: [Type] -> Type) (xs :: [Type]) :: Constraint where
         Extendss p '[]      = ()
-        Extendss p (x : xs) = (Extends p x, Extendss p xs)
+        Extendss p (x : xs) = Extends p x `SeqConstraintUnit` Extendss p xs
```

Comment:
Doesn't this mean that `Extendss` will get stuck if `Extends p x` is not `()` though?
I would expect perhaps `SeqConstraintUnit` to have another equation that creates a pair constraint when it is not a unit?

Reply:
> Doesn't this mean that Extendss will get stuck if Extends p x is not () though?

No, it would be reduced to the form with all unsolvable constraints separated by `SeqConstraintUnit`. Check the playground, the list `[Float, Bool, Double, Int]` would produce such error:
```
   • Could not solve: (SeqConstraintUnit
                          (Extends Proxy Float)
                          (SeqConstraintUnit (Extends Proxy Double) (() :: Constraint)))
```
> I would expect perhaps `SeqConstraintUnit` to have another equation that creates a pair constraint when it is not a unit?

That would not work, if I understand you correctly. Type families do not fall back into the default equation when it is impossible to decide if the argument is reduced into a unit or not.

The custom logic for stucked type families could be done using `{-# INCOHERENT #-}` in type class instances, however, we want to reduce the dictionary cost instead of increasing it.

Reply:
I see, very well.

Reply:
Actually, it may be worth to rename `SeqConstraintUnit`, so it will explain the user what actually happens, like `ReportUnsolved`, e.g.:
```
   • Could not solve: (ReportUnsolved
                          (Extends Proxy Float)
                          (ReportUnsolved (Extends Proxy Double) (() :: Constraint)))
   ```

Reply:
What do you think about that?

Reply:
I think that would be clearer in the error case, indeed.